pkg/cache/v3/linear.go

```diff
@@ -140,18 +140,18 @@ func (cache *LinearCache) respond(value chan Response, staleResources []string)
 
 func (cache *LinearCache) notifyAll(modified map[string]struct{}) {
 	// de-duplicate watches that need to be responded
```

Comment:
It is deduplicating through the map to send a simple reply with n resources rather than n replies with one
On whether to use request.ResourceNames, this would yield a different behavior as:
 - this is not normalized for wildcard parts for instance
 - can target resources which don't exist in the cache, and the respond code might not expect that
 - for eds/rds, we don't want to return all resources but only the ones matching modified

Reply:
I don't think it makes a difference in theory, the request contains all the resources you may need to respond to, so
1. for CDS/LDS and wildcard you need to send all resources for this type
2. for RDS/EDS/* you need the intersection of resource names in `modified` and what is in `request.ResourceNames`